main.cpp

```diff
@@ -137,8 +136,17 @@ struct DoubleType
     DoubleType& multiply(double myNumber);
     DoubleType& divide(double myNumber);
 
+    DoubleType& pow(double myNumber);
+    DoubleType& pow(const IntType& myNumber);
+    DoubleType& pow(const FloatType& myNumber);
+    DoubleType& pow(const DoubleType& myNumber);
+
 private:
     double* value = nullptr;
+    DoubleType& powInternal(double myNumber);
+    DoubleType& powInternal(const IntType& myNumber);
+    DoubleType& powInternal(const FloatType& myNumber);
+    DoubleType& powInternal(const DoubleType& myNumber);
```

Comment:
There should only be one powInternal function

- [ ] remove these extra `powInternal` functions